tests/components/teslemetry/test_config_flow.py

```diff
@@ -86,3 +88,102 @@ async def test_form_errors(hass: HomeAssistant, side_effect, error, mock_test) -
         CONFIG,
     )
     assert result3["type"] is FlowResultType.CREATE_ENTRY
+
+
+async def test_reauth(hass: HomeAssistant, mock_test) -> None:
+    """Test reauth flow."""
+
+    mock_entry = MockConfigEntry(
+        domain=DOMAIN,
+        data=CONFIG,
+    )
+    mock_entry.add_to_hass(hass)
+
+    result1 = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={
+            "source": config_entries.SOURCE_REAUTH,
+            "entry_id": mock_entry.entry_id,
+        },
+        data=CONFIG,
+    )
+
+    assert result1["type"] is FlowResultType.FORM
+    assert result1["step_id"] == "reauth_confirm"
+    assert not result1["errors"]
+
+    with patch(
+        "homeassistant.components.teslemetry.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry:
+        result2 = await hass.config_entries.flow.async_configure(
+            result1["flow_id"],
+            CONFIG,
+        )
+        await hass.async_block_till_done()
+        assert len(mock_setup_entry.mock_calls) == 1
+        assert len(mock_test.mock_calls) == 1
+
+    assert result2["type"] is FlowResultType.ABORT
+    assert result2["reason"] == "reauth_successful"
+    assert mock_entry.data == CONFIG
```

Comment:
Is this not already covered with test_reauth_errors (in particular lines 181 ff.)?

Reply:
Agree and I also don't know if there is convention how to do it. My approach was that if the error-test passes and you get 100% coverage, all should be fine and you need no additional test.

Reply:
A wise @joostlek once told me "I hope you're not just writing tests for coverage", which I usually am, but in this case the success test is intentionally for that reason.

The reason the failure tests also resolve to success is because that's a Home Assistant unwritten best practice.